File: test/acceptance_with_python/_wvhost.py
#                           _       _
# __      _____  __ ___   ___  __ _| |_ ___
# \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
#  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
#   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
#
#  Copyright © 2016 - 2026 Weaviate B.V. All rights reserved.
#
#  CONTACT: hello@weaviate.io
#

"""Resolve Weaviate REST and gRPC endpoints for acceptance tests.

Defaults are localhost:8080 / localhost:50051. WV_TEST_HOST,
WV_TEST_REST_PORT, and WV_TEST_GRPC_PORT override the non-RBAC instance.
WV_TEST_RBAC_HOST, WV_TEST_RBAC_REST_PORT, and WV_TEST_RBAC_GRPC_PORT
override the RBAC/auth instance (defaults localhost:8081 / 50052).

Mirrors test/acceptance_with_go_client/internal/wvhost/wvhost.go so the
Go and Python suites share a single configuration convention.
"""

import os


def host() -> str:
    return os.environ.get("WV_TEST_HOST") or "localhost"


def rest_port() -> int:
    return int(os.environ.get("WV_TEST_REST_PORT") or "8080")


def grpc_port() -> int:
    return int(os.environ.get("WV_TEST_GRPC_PORT") or "50051")


def rest_url() -> str:
    return f"http://{host()}:{rest_port()}"


def rbac_host() -> str:
    return os.environ.get("WV_TEST_RBAC_HOST") or "localhost"


def rbac_rest_port() -> int:
    return int(os.environ.get("WV_TEST_RBAC_REST_PORT") or "8081")


def rbac_grpc_port() -> int:
    return int(os.environ.get("WV_TEST_RBAC_GRPC_PORT") or "50052")


def rbac_rest_url() -> str:
    return f"http://{rbac_host()}:{rbac_rest_port()}"

File: test/acceptance_with_go_v6_client/internal/wvhost/wvhost.go
//                           _       _
// __      _____  __ ___   ___  __ _| |_ ___
// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
//
//  Copyright © 2016 - 2026 Weaviate B.V. All rights reserved.
//
//  CONTACT: hello@weaviate.io
//

// Package wvhost resolves the Weaviate REST and gRPC endpoints for acceptance
// tests. Defaults are localhost:8080 / localhost:50051 / localhost:6060;
// WV_TEST_HOST, WV_TEST_REST_PORT, WV_TEST_GRPC_PORT and WV_TEST_DEBUG_PORT
// override host and port independently.
//
// The v6 client takes host and port separately, so ports are exposed as ints
// next to the "host:port" forms the REST endpoints still use.
package wvhost

import (
	"os"
	"strconv"
)

func Host() string {
	if h := os.Getenv("WV_TEST_HOST"); h != "" {
		return h
	}
	return "localhost"
}

func REST() string {
	return Host() + ":" + strconv.Itoa(RESTPort())
}

func GRPC() string {
	return Host() + ":" + strconv.Itoa(GRPCPort())
}

func Debug() string {
	return Host() + ":" + strconv.Itoa(DebugPort())
}

func RESTPort() int {
	return port("WV_TEST_REST_PORT", 8080)
}

func GRPCPort() int {
	return port("WV_TEST_GRPC_PORT", 50051)
}

func DebugPort() int {
	return port("WV_TEST_DEBUG_PORT", 6060)
}

func port(env string, def int) int {
	p, err := strconv.Atoi(os.Getenv(env))
	if err != nil {
		return def
	}
	return p
}
